nncf/torch/dynamic_graph/io_handling.py

```diff
@@ -229,6 +229,20 @@ def __init__(self, forward_args: Tuple, forward_kwargs: Dict):
     def get_forward_inputs(self, device: str = None) -> Tuple[Tuple, Dict]:
         return self._forward_args, self._forward_kwargs
 
+    @classmethod
+    def from_example_input(cls, example_input: Any) -> "ExactInputsInfo":
```

Comment:
I really think we shouldn't have an `example_input` entity with a non-trivial logic of turning it into forward args and kwargs. 

In torch's jit.trace function (https://pytorch.org/docs/stable/generated/torch.jit.trace.html) there are now both `example_input` and `example_kwarg_inputs` parameters. Their handling of these parameters is "either/or", which basically maps onto your current logic, so your solution is not much worse than theirs. But I highly suspect that the only reason they had to do this is because they designed this API poorly in the first place (see [complaint](https://discuss.pytorch.org/t/torch-jit-trace-example-inputs-how-to-set-specified-parameters/175005)) and they needed to preserve backward compatibility. I propose not to repeat the torch mistakes and switch the understanding of the `example_inputs` in the Torch case from a `Union[Tuple, Dict]` to an intuitive `(args, kwargs)` tuple, so the `from_example_input` body would look like:
```python3
PTExampleInput = Tuple[Tuple[Any, ...], Dict[str, Any]]

@classmethod
def from_example_input(cls, example_input: PTExampleInput):
    args, kwargs = example_input
    return ExactInputsInfo(args, kwargs)
```
If not, then at least the `PTExampleInput` should be defined as a typedef like above with the currently applicable type, and the corresponding type used everywhere where such an object is expected, e.g. for the current state:
```python3
PTExampleInput = Union[Tuple[Any, ...], Dict[str, Any]]  # use this below, also in .wrap_model signature and in tests
@classmethod
def from_example_input(cls, example_input: PTExampleInput):
    ... # same as your current variant
```

Reply:
The general idea to be align with openvino https://docs.openvino.ai/2023.1/openvino_docs_OV_Converter_UG_prepare_model_convert_model_Convert_Model_From_PyTorch.html. Could you make proposal to OpenVINO team?

Reply:
If they made it their API already, there is little they can do at this point in time.